src/x509.c

```diff
@@ -7202,6 +7214,402 @@ WOLFSSL_X509_CRL* wolfSSL_d2i_X509_CRL(WOLFSSL_X509_CRL** crl,
+            if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+                return WOLFSSL_FAILURE;
+            }
+
+            if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+                return WOLFSSL_FAILURE;
+            }
+        #endif
+            revoked = revoked->next;
+        }
+    }
+    else {
+        if (wolfSSL_BIO_write(bio, "No Revoked Certificates.\n",
+                       (int)XSTRLEN("No Revoked Certificates.\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+
+    return WOLFSSL_SUCCESS;
+}
+
+#ifndef NO_ASN_TIME
+/* print out the last/next update times in human readable
+ * format for use with wolfSSL_X509_CRL_print()
+ * return WOLFSSL_SUCCESS on success
+ */
+static int X509CRLPrintDates(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl,
+        int indent)
+{
+    char tmp[MAX_WIDTH]; /* buffer for XSNPRINTF */
+
+    XSNPRINTF(tmp, MAX_WIDTH, "%*s%s", indent, "",
+            "Last Update: ");
+
+    if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (crl->crlList->lastDate[0] != 0) {
+        if (GetTimeString(crl->crlList->lastDate, ASN_UTC_TIME,
+            tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+            if (GetTimeString(crl->crlList->lastDate, ASN_GENERALIZED_TIME,
+            tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+                WOLFSSL_MSG("Error getting last update date");
+                return WOLFSSL_FAILURE;
+            }
+        }
+    }
+    else {
+        XSTRNCPY(tmp, "Not Set", sizeof(tmp)-1);
+    }
+    tmp[sizeof(tmp) - 1] = '\0'; /* make sure null terminated */
+    if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    XSNPRINTF(tmp, MAX_WIDTH, "%*s%s", indent, "",
+            "Next Update: ");
+
+    if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (crl->crlList->nextDate[0] != 0) {
+        if (GetTimeString(crl->crlList->nextDate, ASN_UTC_TIME,
+            tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+            if (GetTimeString(crl->crlList->nextDate, ASN_GENERALIZED_TIME,
+            tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+                WOLFSSL_MSG("Error getting next update date");
+                return WOLFSSL_FAILURE;
+            }
+        }
+    }
+    else {
+        XSTRNCPY(tmp, "Not Set", sizeof(tmp)-1);
+    }
+    tmp[sizeof(tmp) - 1] = '\0'; /* make sure null terminated */
+    if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    return WOLFSSL_SUCCESS;
+}
+#endif
+
+#if !defined(NO_BIO) && defined(XSNPRINTF)
+int wolfSSL_X509_CRL_print(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl)
```

Comment:
lets add a comment to all new functions added, at the minimum what the expected return value for success is